druid/src/core.rs

```diff
@@ -106,6 +106,25 @@ pub(crate) struct WidgetState {
     pub(crate) viewport_offset: Vec2,
 
     // TODO: consider using bitflags for the booleans.
+    // Any child of this widget changed the disabled state and should either receive
+    // LifeCycle::DisabledChanged or InternalLifeCycle::RouteDisabledChanged
+    pub(crate) children_disabled_changed: bool,
+
+    // Any of our ancestors are disabled. Therefore we are also disabled.
+    pub(crate) ancestor_disabled: bool,
+
+    // explicitly disabled through the usage of EventCtx::set_disabled(true) on this widget.
+    pub(crate) is_explicitly_disabled: bool,
+
+    // A buffer for the value of EventCtx::set_disabled() before this widget receives
+    // LifeCycle::DisabledChanged or InternalLifeCycle::RouteDisabledChanged
+    pub(crate) is_explicitly_disabled_new: bool,
+
+    // This widget get focused automatically.
+    // We need to store this here since the widget local focus-chain is cleared when the widget is
+    // disabled
```

Comment:
```suggestion
    // `true` if this widget participates in the focus chain.
    // We store this as a bool because we need to know how to update the focus chain
    // when disabled/enabled state changes.
```